app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt

```diff
@@ -6,11 +6,54 @@
 
 package mozilla.lockbox.presenter
 
+import mozilla.lockbox.adapter.AppVersionSettingConfiguration
+import mozilla.lockbox.adapter.SectionedAdapter
+import mozilla.lockbox.adapter.SettingCellConfiguration
+import mozilla.lockbox.adapter.TextSettingConfiguration
+import mozilla.lockbox.adapter.ToggleSettingConfiguration
 import mozilla.lockbox.flux.Presenter
+import mozilla.lockbox.BuildConfig
 
-interface SettingViewProtocol
+interface SettingView {
+    fun updateSettingList(
+        settings: List<SettingCellConfiguration>,
+        sections: List<SectionedAdapter.Section>
+    )
+}
 
-class SettingPresenter(view: SettingViewProtocol) : Presenter() {
+class SettingPresenter(val view: SettingView) : Presenter() {
+    private val versionNumber = BuildConfig.VERSION_NAME
     override fun onViewReady() {
+        val settings = listOf(
+            ToggleSettingConfiguration(
+                title = "Unlock with fingerprint",
+                toggle = false
+            ),
+            TextSettingConfiguration(
+                title = "Auto lock",
+                detailText = "5 minutes"
+            ),
+            ToggleSettingConfiguration(
+                title = "Autofill",
+                subtitle = "Let Firefox Lockbox fill in logins for you",
+                toggle = false
+            ),
+            ToggleSettingConfiguration(
+                title = "Send usage data",
+                subtitle = "Mozilla strives to only collect what we need to provide and improve Firefox for everyone. ",
+                buttonTitle = "Learn more",
+                toggle = true
+            ),
+            AppVersionSettingConfiguration(
+                text = "App Version: $versionNumber"
```

Comment:
This bothers me: there isn't any use of `R.string` so far, and if we're dealing with Strings in this `Presenter` then we need to have a `Context.getResources()`, which goes against the separation of view from presenter.

Suggest you switch the configuration objects to use `@StringRes Int`, with a `res/values/settings.xml` strings file.